app/models/pawn.rb

```diff
@@ -1,10 +1,38 @@
 class Pawn < Piece
-  
   def self.get_image(color)
     if color == "white"
       return "chess_piece_pawn_white.png"
     elsif color == "black"
       return "chess_piece_pawn_black.png"
     end
   end
+
+  def valid_move?(x,y)
+    @object = self.game.pieces.where(position_x: x, position_y: y)[0]
```

Comment:
there are a few checks like `is_on_board?` or if its a piece of opposite color which can be done in the beginning of the method. i.e. if x&y are not on board, there is no need to calculate all the valid moves, right? the same is if it is my own piece at the destination.. 

--> lets add those checks in the beginning and return eliminate them there